karrot/offers/receivers.py

```diff
@@ -13,4 +14,12 @@ def offer_saved(sender, instance, created, **kwargs):
         conversation = Conversation.objects.get_or_create_for_target(offer)
         conversation.join(offer.user)
 
-        notify_members_about_new_offer(offer)
+        def notify():
+            notify_members_about_new_offer(offer)
+
+        # offer saving is normally done in a transaction so as to include the images
+        # we only want to trigger the notification after this transaction is complete
+        if transaction.get_connection().in_atomic_block:
```

Comment:
Docs say: 

> If you call on_commit() while there isn’t an active transaction, the callback will be executed immediately.

Did you try it without the check?

Reply:
Yup, good catch, I removed the check (I confirmed it still ran when the receiver wasn't called in a transaction, but also just read the code, it's quite clear).